backend-plugin/src/main/java/com/redhat/jenkins/nodesharingbackend/Api.java

```diff
@@ -249,38 +251,68 @@ public void doReportWorkload(@Nonnull final StaplerRequest req, @Nonnull final S
         final ConfigRepo.Snapshot config = pool.getConfig(); // Fail early when there is no config
 
         final ReportWorkloadRequest request = Entity.fromInputStream(req.getInputStream(), ReportWorkloadRequest.class);
+        final ExecutorJenkins executor = config.getJenkinsByUrl(request.getExecutorUrl());
+
+        final ArrayList<ReservationTask> processed = new ArrayList<>();
+
+        // First collect what is executed on nodes already
+        for (Node n : Jenkins.getInstance().getNodes()) {
+            for (Executor e : n.toComputer().getExecutors()) {
+                if (e.getCurrentExecutable() instanceof ReservationTask.ReservationExecutable) {
+                    ReservationTask rs = ((ReservationTask.ReservationExecutable) e.getCurrentExecutable()).getParent();
```

Comment:
There is a shortcut for this: `for (ShareableNode sn: ShareableNode.getAll()) { sn.getComputer().getReservation }`